File: create_item.go
// Random Go struct generated on 2025-08-08
package main

import (
    "encoding/json"
    "fmt"
    "time"
)

type GetItem struct {
    Name      string    `json:"name"`
    Value     int       `json:"value"`
    Active    bool      `json:"active"`
    CreatedAt time.Time `json:"created_at"`
}

func NewGetItem(name string, value int) *GetItem {
    return &GetItem{
        Name:      name,
        Value:     value,
        Active:    true,
        CreatedAt: time.Now(),
    }
}

func (s *GetItem) UpdateValue(newValue int) {
    s.Value = newValue
}

func (s *GetItem) ToJSON() ([]byte, error) {
    return json.Marshal(s)
}

func main() {
    obj := NewGetItem("sample", 29)
    data, _ := obj.ToJSON()
    fmt.Println(string(data))
}

File: create_item.c
// Random C struct generated on 2025-09-15
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

typedef struct {
    char name[100];
    int value;
    int active;
    long timestamp;
} GetResult;

GetResult* create_getresult(const char* name, int value) {
    GetResult* obj = malloc(sizeof(GetResult));
    if (obj == NULL) return NULL;
    
    strncpy(obj->name, name, sizeof(obj->name) - 1);
    obj->name[sizeof(obj->name) - 1] = '\0';
    obj->value = value;
    obj->active = 1;
    obj->timestamp = 1849837723;
    
    return obj;
}

void update_getresult_value(GetResult* obj, int new_value) {
    if (obj != NULL) {
        obj->value = new_value;
    }
}

void print_getresult(const GetResult* obj) {
    if (obj != NULL) {
        printf("Name: %s, Value: %d, Active: %d\n", 
               obj->name, obj->value, obj->active);
    }
}

void free_getresult(GetResult* obj) {
    if (obj != NULL) {
        free(obj);
    }
}

int main() {
    GetResult* obj = create_getresult("sample", 32);
    print_getresult(obj);
    free_getresult(obj);
    
    return 0;
}
